src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -226,9 +226,10 @@ protected override void OnNextCore(Block block)
             }
 
             // Announce the blocks to each of the peers.
-            IEnumerable<BlockStoreBehavior> behaviours = peers.Select(s => s.Behavior<BlockStoreBehavior>());
+            List<BlockStoreBehavior> behaviours = peers.Select(s => s.Behavior<BlockStoreBehavior>())
```

Comment:
Select will throw if peers collection is modified during the foreach which is used in the select
https://github.com/Microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs

You can rewrite this to avoid using linq and avoid having a problem like mentioned above

Reply:
I think it is OK: Linq uses the GetEnumerator() function of the Enumerable, which is for ConcurrentHashSet the following
![image](https://user-images.githubusercontent.com/4638821/42923798-529760b2-8b1f-11e8-9982-9532e0a6f022.png)

Reply:
Ah, yes, peer collection uses ConcurrentHashSet inside. Then ok, it should be good.